builtin/core.wren

```diff
@@ -21,6 +21,16 @@ class Sequence {
     }
     return true
   }
+
+  reduce(acc, f) {
+    for (element in this) {
+      acc = f.call(acc, element)
+    }
+    return acc
+  }
+
+  reduce(f) { this[1..-1].reduce(this[0], f) }
```

Comment:
I considered putting it down into the List class, but that felt a bit wrong as well. I guess it could be implemented without the indexing, but I don't think it's worth the complexity.